src/Model/PageTypeGo.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Model;
+using AutoRest.Core.Utilities;
+using AutoRest.Extensions.Azure;
+using AutoRest.Go;
+using System;
+using System.Linq;
+
+namespace AutoRest.Go.Model
+{
+    /// <summary>
+    /// Represents a response from a pageable operation.
+    /// </summary>
+    public class PageTypeGo : CompositeTypeGo
+    {
+        /// <summary>
+        /// Creates a new pageable type for the specified response type.
+        /// </summary>
+        /// <param name="method">The method that will return a pageable type.</param>
+        public PageTypeGo(MethodGo method) : base(CodeNamerGo.Instance.GetPageTypeName(method))
+        {
+            if (!method.IsPageable)
+            {
+                throw new InvalidOperationException($"method {method.Owner}.{method.Name} is not a pageable operation");
+            }
+
+            CodeModel = method.CodeModel;
+            ContentType = (CompositeTypeGo)method.ReturnType.Body;
+            ElementType = ContentType.Properties.Where(p => p.ModelType is SequenceTypeGo).FirstOrDefault().ModelType.Cast<SequenceTypeGo>().ElementType;
+            Documentation = $"Contains a page of {ElementType.Name} values.";
+            PreparerNeeded = !method.NextMethodExists(CodeModel.Methods.Cast<MethodGo>());
+
+            var pageableExtension = method.Extensions[AzureExtensions.PageableExtension] as Newtonsoft.Json.Linq.JContainer;
+            NextLink = CodeNamerGo.Instance.GetPropertyName((string)pageableExtension["nextLinkName"]);
```

Comment:
> NextLink = CodeNamerGo.Instance.GetPropertyName((string)pageableExtension["nextLinkName"]); [](start = 11, length = 92)

I would look into other generators to know if there is a fancier way of accessing this value

Reply:
This was lifted from the C# generator.